src/ray/core_worker/transport/normal_task_submitter.cc

```diff
@@ -724,12 +721,9 @@ Status NormalTaskSubmitter::CancelTask(TaskSpecification task_spec,
       for (auto spec = scheduling_tasks.begin(); spec != scheduling_tasks.end(); spec++) {
         if (spec->TaskId() == task_spec.TaskId()) {
           scheduling_tasks.erase(spec);
-
-          if (scheduling_tasks.empty()) {
```

Comment:
We shouldn't remove this `if` check, we should only cancel worker lease if there is no scheduling task.

Reply:
Actually removed this because CancelWorkerLeaseIfNeeded already checks if its empty and early returns if it's not empty